app/src/main/java/com/rob729/quiethours/util/EndAlarm.kt

```diff
@@ -4,51 +4,28 @@ import android.app.NotificationManager
 import android.app.PendingIntent
 import android.content.Context
 import android.content.Intent
-import android.graphics.Color
 import android.media.AudioManager
-import androidx.core.app.NotificationCompat
 import androidx.work.Worker
 import androidx.work.WorkerParameters
 import com.rob729.quiethours.Activity.SplashScreen
-import com.rob729.quiethours.R
 
 class EndAlarm(appContext: Context, workerParams: WorkerParameters) :
     Worker(appContext, workerParams) {
-    private val b = "422"
 
     override fun doWork(): Result {
-
         val profileName = inputData.getString("Profile_Name")
-        val notificationManager =
-            applicationContext.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
-
-        val audioManager =
-            applicationContext.getSystemService(Context.AUDIO_SERVICE) as AudioManager
+        val notificationManager = applicationContext.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        val audioManager = applicationContext.getSystemService(Context.AUDIO_SERVICE) as AudioManager
```

Comment:
Move audioManager to the application class and use the reference from there only. Remove the audioManager class from everywhere it is initialized before.

Reply:
@rob729 I am a bit confused, while initializing sharedprefs and work manager we used an object and then passed context from application class in init......
So shall I place audio manager also in work manager helper or what?

Reply:
No, just create a public variable in the application class and lazy initialize it.